src/main/java/org/shipkit/auto/version/NextVersionPicker.java

```diff
@@ -30,13 +34,18 @@ String pickNextVersion(Optional<Version> previousVersion, VersionConfig config,
         }
 
         if (!config.getRequestedVersion().isPresent()) {
-            String tag = runner.run("git", "describe", "--tags");
-            Pattern pattern = Pattern.compile(config.getTagPrefix() + "\\d+\\.\\d+\\.\\d+");
-            Matcher matcher = pattern.matcher(tag);
+            String tag = runner.run("git", "describe", "--tags").trim();
             String result;
-            if (matcher.find()) {
-                result = matcher.group().substring(config.getTagPrefix().length());
-                explainVersion(log, result, "deducted version based on tag: '" + config.getTagPrefix() + result + "'");
+            if (isVersionTag(tag, config.getTagPrefix()) && isSupportedVersion(tag.substring(config.getTagPrefix().length()))) {
+                result = tag.substring(config.getTagPrefix().length());
+                explainVersion(log, result, "deducted version based on tag: '" + tag + "'");
+            } else if (isVersionTag(tag, config.getTagPrefix()) && isSnapshot(tag, config.getTagPrefix())){
```

Comment:
Let's push the logic of `isVersionTag` to `isSnapshot` method. I think it will be cleaner. It's OK to do this in a separate PR. Bonus: we might do the same with `isSupportedVersion`.